cmd/kube-controller-manager/controller-manager.go

```diff
@@ -46,10 +51,35 @@ func main() {
 	util.InitLogs()
 	defer util.FlushLogs()
 
-	verflag.PrintAndExitIfRequested()
+	//Functions to start and stop this daemon.
+	startRC := func(leaseUserInfo *ha.LeaseUser) bool {
+		leaseUserInfo.Running = true
+		if err := s.Run(pflag.CommandLine.Args()); err != nil {
+			fmt.Fprintf(os.Stderr, "%v\n", err)
+			glog.Infof("EXITING NOW ! Killed")
+			os.Exit(1)
+		}
+		return true
+	}
+
+	endRC := func(leaseUserInfo *ha.LeaseUser) bool {
+		glog.Infof("Hard-exiting the replication controller process now !")
+
+		//Even though we're exiting, we should set this flag before dying just for completeness.
+		leaseUserInfo.Running = false
+		os.Exit(0)
+
+		//this should never be reached
+		return true
+	}
+
+	//This starts a thread that continues running.
+	ha.RunHA(s.Kubeconfig, s.Master, startRC, endRC, "ha.cm.lock")
```

Comment:
Should it?  This is the controller-manager.  Is there value in having the lock name configurable for the kcm?  I was thinking we would want to hard-code the lock name in the processes so there's no chance of a configuration snaffu leading to issues